### ISSUE TITLE
Too many ASVs occurring only once across samples - 16S amplicon

### ISSUE THREAD

**Issue by Mathildebd** [benjjneb/dada2#1233]
Hi,
I am repeating here a comment I wrote to another issue #1194  yesterday (sorry if this is annoying!)
I am adding here some better description and more information.
I have a 16S amplicon dataset which I processed through the Dada2 pipeline (non-pooling), which these main settings:

`out <- filterAndTrim(fwd=file.path(path, fnFs), filt=file.path(filtpath, fnFs),
                     rev=file.path(path, fnRs), filt.rev=file.path(filtpath, fnRs),
                     truncLen=c(250,200), maxEE=2, truncQ=2,minLen = 50, trimLeft = c(19,0), trimRight = c(0,20), maxN = 0, rm.phix=TRUE,
                     compress=TRUE, verbose=TRUE, multithread=FALSE)`

The trimming is cutting away the primers (tested prior to the main filtering step) and the `truncLen` is based on a visual assessment of the quality plots.
I am obtaining a tremendous amount of 'unique' ASVs in my samples - all 93 samples are from a similar environments, forest soil samples, north Germany.

I find in total 16,978 ASVs, but 13,444 only occurs in ONE sample.
The number is 14,981 if I excluded ASVs only present in two samples - leaving 1,997 ASVs for the further analysis.
I find this number quite extreme - and it makes me uneasy about the validity of the ASV table...

Is there something known to you which can cause exactly this problem?

I find some of these low frequency (unique) ASVs to account for quite a proportion of the community in the samples they occur in, see plot below.
Furthermore, because they are _that_ many they actually account for quite a bit of the relative abundance (unavoidable).
Some key numbers:
The community which is left after filtering (i.e. 1,997 ASVs in total) represent on average 59.7 +/- 12.1% of the original abundance (minimum case being 31.7% left of the community) - this also makes me unease.

![ASV_count_vs_sum_relativeAbundance](https://user-images.githubusercontent.com/9782235/102468375-b9488b80-4051-11eb-896f-a700a668c47a.png)

**Comment by benjjneb** [comment]
Usually it's practice to open a new issue if your problem isn't an exact copy of a previous issue.

Can. you descibe a little bit more here on the technical and environmental qualities of your study? I.e. what amplicon and environment are you sequencing here?

**Comment by Mathildebd** [comment]
Hi - Thank you for your fast reply so close to the holidays.

Let me try to give some more details. Like I mentioned the samples are from soil (so we expect the diversity to be high as it is in soil environments) - 9 sites in Northern Germany beech forest, with two manipulation treatments per site (warming/cooling) and a control. Samples were taking in 3 biological replicates within a few meters apart -total 93 samples.

16S EMP primers (515F/806R) for the V4 amplicon.

The quality of the sequences is not great - A lot is lost in the filtering step see track-table attached and quality plots for a random selection of samples:
[16S_track.txt](https://github.com/benjjneb/dada2/files/5723229/16S_track.txt)
![R1](https://user-images.githubusercontent.com/9782235/102757130-21a8ad00-4371-11eb-933d-04a33a702b60.png)
![R2](https://user-images.githubusercontent.com/9782235/102757133-22d9da00-4371-11eb-8444-9e618d047e92.png)

In the end ca. 30% (~25,000) of the input reads (~90,000 per sample) are saved after the last step.

When evaluation the community in the samples both sites and biological replicates are to some extend captured - though there are also a significant problem with _one_ very deep split between two groups of  (seemingly random) samples, see below.
[Sample_dist._All.pdf](https://github.com/benjjneb/dada2/files/5723550/Sample_dist._All.pdf)
[Sample_dist._bd.pdf](https://github.com/benjjneb/dada2/files/5723551/Sample_dist._bd.pdf)

It is also evident that the pattern is robust to the filtering of rare ASVs (sample design may be a bit better captured by the large dataset, as expected).
I cannot explain the deep split, it is not related to richness (no. of ASVs in the samples) and I haven't identified any unique taxonomic signature in these samples.

Finally, I looked into the taxonomic classification (below). If assuming the many rare ASVs are somehow noise or not representing anything real, one could expect that they cannot be classified. However, when filtering out singletons and doubletons (here defined by number of occurrences in samples) a lot of deeper classifications are lost (e.g. in the 'raw' dataset 166 genera are represented, in the filtered only 35 genera). This suggest that they _do_ represent real taxa in the samples.

![tax_summary](https://user-images.githubusercontent.com/9782235/102764032-ced3f300-437a-11eb-8b67-b1f42c7f97a4.png)

I would greatly appreciate any thought you have on this data. I am currently considering applying a stronger `TruncLen` filter (potentially jeopardizing the merging)... other than that I don't have any good idea as to what to do next.

**Comment by benjjneb** [comment]
Given the non-ideal quality profiles of your reads, I would suggest using an external program to merge the reads first, and then using DADA2 to denoise them (as single-end reads). If you do that, does it ameliorate these patterns you are concerned by?

Note: This is not usually recommended because different relationships between the quality score and error rates can occur in the overlapping and non-overlapping parts of merged reads. But the 2x250 V4 sequencing (without primers sequences) is essentially completely overlapping, so that issue can be ignored.

**Comment by Mathildebd** [comment]
Thanks again for the fast reply, I was away from my desk a few days due to Christmas.

I have now tried to merge the reads using Usearch (Edgar) applying no filter whatsoever (e.g. min length, maxEE etc.). This gives me _one_ fastq-file (I haven't found a way to get one file per sample). I thought theoretically this could work fine (though I might run into trouble when running the `makeSequenceTable()` - I imaging it might be needing some sample-identifiers...). 
But, I haven't reached that point yet since running the `dada()` itself is causing problems, i.e. it has been running for 24h now, so I think something is wrong. 
FYI I used a subset sample to estimate the error rate: 

> (8190058 total bases in 31643 reads from 1 samples will be used for learning the error rates)

And I can also say that 53.59% was merged using Usearch, 43.12% failed due to too many mismatched in the overlap (default 5; I might increase this threshold when/if I can get this workflow to work).

Can you advise me one 1) is it possible to use this one merged file in the dada2 pipeline? or 2) do you know a way/program to merge the files and keep individual samples as separate files? 
I am running on windows, from where I can run Usearch in a cygwin terminal. My bioinformatic skills are pretty much limited to usearch and dada2.

Best wishes to you for the new year!

**Comment by benjjneb** [comment]
> IGNORE - I have found a solution and will update this comment when I know how it all turned out!

@Mathildebd I was writing a response but just saw this in your comment. Please re-comment if you still have a problem (or if you found a solution!).

**Comment by Mathildebd** [comment]
@benjjneb I hope I did it right now - editing the comment above.
